File: UISegmentedControlObjC/UISegmentedControlObjC/RectangleView.m
//
//  RectangleView.m
//  UISegmentedControlObjC
//
//  Created by Hugo Flores Perez on 3/25/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

#import "RectangleView.h"

@interface ShapeDrawer ()
@property UIBezierPath *path;
- createRectangle;
@end

@implementation RectangleView

// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    [self createRectangle];
    
    [UIColor.orangeColor setFill];
    [self.path fill];
}

@end

File: UISegmentedControlSwift/UISegmentedControlSwift/RectangleView.swift
//
//  RectangleView.swift
//  UISegmentedControlSwift
//
//  Created by Field Employee on 3/25/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

import UIKit

class RectangleView: Demoview {

    override func draw(_ rect: CGRect) {
        self.createRectangle()
        
        // Specify the fill color and apply it to the path.
        UIColor.orange.setFill()
        path.fill()
        
        // Specify a border (stroke) color.
        UIColor.purple.setStroke()
        path.stroke()
    }

}
